docs/earth-config/earth-config.md

```diff
@@ -0,0 +1,34 @@
+# Earthly config file
+
+Global configuration values for earth can be stored on disk in the configuration file.
+
+By default, earth reads the configuration file `~/.earthly/config.yaml`; however, it can also be
+overridden with the `--config` command flag option.
+
+Currently only https authentication configuration is supported by the config file.
+
+## Format
+
+The earthly config file is a [yaml](https://yaml.org/) formatted file that looks like:
+
+```yaml
+git:
+    <site>:
+        auth: https
+        user: <username>
+        password: <password>
+    <site2>:
+        ...
+```
```

Comment:
Might be worth also adding a `github.com` example just below this format.

Reply:
excellent point